droidmediacodec.cpp

```diff
@@ -454,9 +454,22 @@ class DroidMediaCodecBuilder {
 #endif
 
       if (!strcmp (mime, android::MEDIA_MIMETYPE_VIDEO_AVC)) {
-        format->setInt32("profile", OMX_VIDEO_AVCProfileBaseline);
-        int32_t level = android::ACodec::getAVCLevelFor(width, height, frames, bitrate);
-        format->setInt32("level", level);
+        if (m_enc->codec_specific.h264.profile) {
+          format->setInt32("profile", m_enc->codec_specific.h264.profile);
+        } else {
+          format->setInt32("profile", OMX_VIDEO_AVCProfileBaseline);
+        }
+
+        if (m_enc->codec_specific.h264.level) {
+          format->setInt32("level", m_enc->codec_specific.h264.level);
+        } else {
+          int32_t level = android::ACodec::getAVCLevelFor(width, height, frames, bitrate);
```

Comment:
Should we give the profile parameter to this function? That is the last parameter after bitrate which we currently don't add it in which case it will be using OMX_VIDEO_AVCProfileBaseline as default value.

Reply:
Fixed. Something strange is happening there. Event though `getAVCLevelFor()` takes `OMX_VIDEO_AVCPROFILEEXTTYPE` as a parameter, it  casts to int32 and treats it like `OMX_VIDEO_AVCPROFILETYPE` http://androidxref.com/9.0.0_r3/xref/frameworks/av/media/libstagefright/ACodec.cpp#4210.